### PR TITLE
Composed MC label to encode Track, Event and Source IDs

### PR DESCRIPTION
**Pull request by shahor02**

### CODE REVIEW

**Comment by shahor02** [comment]
@sawenzel : here is it extracted to separate PR

File: DataFormats/simulation/include/SimulationDataFormat/MCCompLabel.h [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef ALICEO2_MCCOMPLABEL_H
+#define ALICEO2_MCCOMPLABEL_H
+
+#include <Rtypes.h>
+#include <iomanip>
+#include <ios>
+#include <iostream>
```
Comment (sawenzel) [inline DataFormats/simulation/include/SimulationDataFormat/MCCompLabel.h]: can we avoid having this expensive include in a header? consider moving the print function to .cxx

File: DataFormats/simulation/include/SimulationDataFormat/MCCompLabel.h [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  bool isEmpty() const { return mLabel == NotSet; }
+
+  // check if label was assigned with non-negaive trackID
+  bool isPosTrackID() const { return getTrackID() >= 0; }
+  
+  // conversion op-r
+  operator ULong64_t() const { return mLabel; }
+  // invalidate
+  void unset() { mLabel = NotSet; }
+  void set(int trackID, int evID, int srcID)
```
Comment (sawenzel) [inline DataFormats/simulation/include/SimulationDataFormat/MCCompLabel.h]: should we check whether the input parameters satisfy the space/bit contraints expected by this class?

File: DataFormats/simulation/include/SimulationDataFormat/MCCompLabel.h (hunk 1 above)
Comment (shahor02) [reply DataFormats/simulation/include/SimulationDataFormat/MCCompLabel.h]: ok

**Comment by sawenzel** [comment]
@wiechula , @a-mathis : Can you please take a quick look if this label encoding satisfied your requirements?

File: DataFormats/simulation/include/SimulationDataFormat/MCCompLabel.h (hunk 2 above)
Comment (shahor02) [reply DataFormats/simulation/include/SimulationDataFormat/MCCompLabel.h]: doing this per label  basis is unnecessary overhead: evID and srcID is the same for many digits, can be checked (it is in the ITSdigitization code in other PR) once received from the file manager. For the trackID, it is int which may accept also negative values, I'll add static_assert to make sure the field for trackID is exactly sizeof(int)

File: DataFormats/simulation/include/SimulationDataFormat/MCCompLabel.h [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+  ULong64_t mLabel = NotSet; ///< MC label encoding MCtrack ID and MCevent origin
+
+ public:
+
+  // number of bits reserved for MC track ID, DON'T modify this, sine the
+  // track ID might be negative
+  static constexpr int nbitsTrackID = 32;
+  static constexpr int nbitsEvID = 19;    // number of bits reserved for MC event ID
+  static constexpr int nbitsSrcID = 8;    // number of bits reserved for MC source ID
```
Comment (sawenzel) [inline DataFormats/simulation/include/SimulationDataFormat/MCCompLabel.h]: idea: we could potentially try to make MCCompLabel configurable by passing these things as template parameters?

File: DataFormats/simulation/src/MCCompLabel.cxx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "SimulationDataFormat/MCCompLabel.h"
+#include <iomanip>
+#include <ios>
+#include <iostream>
+#include <assert.h>
```
Comment (sawenzel) [inline DataFormats/simulation/src/MCCompLabel.cxx]: is this the right header for static_assert? In any case, it should be `<cassert>` ... the codechecker complains.

**Comment by wiechula** [comment]
For TPC this looks fine. Thanks for this initiative!

**Comment by sawenzel** [comment]
Will be merged via #540 where merge conflicts have been resolved (via rebase).

**Pull request closed without merging**